subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBase.java

```diff
@@ -524,7 +527,7 @@ public long getLastEventOrderedId() {
         return it.hasNext() ? ((SubscriptionBaseTransitionData) it.next()).getTotalOrdering() : -1L;
     }
 
-    public List<SubscriptionBillingEvent> getBillingTransitions() {
+    public List<SubscriptionBillingEvent> getBillingTransitions(final Catalog catalog) throws CatalogApiException {
```

Comment:
Worth adding small unit tests just for this (since it's the core of the change and the logic is... subtle 😅)?

Reply:
yep... see https://github.com/killbill/killbill/commit/87ea0ec55292e73d08fc19f1c941dc180da8a3d3#diff-52efa6b66884d11316c9bde8c72f90e4R45